addons/website/views/website_templates.xml

```diff
@@ -271,7 +271,9 @@
         <attribute name="t-if">website.user_id != user_id</attribute>
     </xpath>
     <xpath expr="//header//ul[@id='top_menu']/li[hasclass('divider')]" position="before">
-        <t t-foreach="website.menu_id.child_id" t-as="submenu">
+        <!-- Bypass ir.rule if no group on menus -->
+        <t t-set="_website" t-value="website if request.env['website.menu'].sudo().search(website.website_domain()).group_ids else website.sudo()"/>
```

Comment:
If we must come to that hack... I would try to include the group_ids != False in your search domain if possible

Reply:
Hopefully replaced by select_distinct_from_where_not_null merge queries.
Kept this commit + its revert to keep the diff if needed